Upgrade Periscope

gurmukhp opened the pull request:
- Migrate from SystemJS to WebPack
- Triage View now working
- Clang formatted all files
- Replaced deprecated Firebase Auth API

mhevery commented on src/app/+sync/sync.component.ts:
```diff
@@ -5,7 +5,6 @@ import {Digest, GithubStore} from '../github/store';
 import {Event} from '../github/v3';
 
 @Component({
-  moduleId: module.id,
```
Why did you delete these?

gurmukhp replied:
The [migration guide](https://github.com/angular/angular-cli/wiki/Upgrading-from-Beta.10-to-Beta.12) suggested to do this:

"Remove all mention of moduleId: module.id. In webpack, module.id is a number but Angular expect a string."

mhevery replied:
@hansl can probably comment more, but I think removing it will make it only work with web-pack, which I guess is fine for us, so let's keep it out.

mhevery commented on src/app/+triage-pr/triage-pr.component.ts:
```diff
@@ -2,10 +2,9 @@ import {Component} from '@angular/core';
 import {PrBoardService, Staleness} from '../pr-board.service';
 
 @Component({
-  moduleId: module.id,
```
same here, why delete?

mhevery commented on src/app/github/store.ts:
```diff
@@ -4,19 +4,19 @@ import 'rxjs/add/operator/partition';
 import 'rxjs/add/operator/do';
```
This whole storage file should be rewritten (not now but later.) The current file tries to process the incoming queue from the github in the browser. Since we now have server which does that for us, this is unnecessary, and server should be done all future processing. 

Just FYI

mhevery commented on src/polyfills.ts:
```diff
@@ -0,0 +1,19 @@
+// This file includes polyfills needed by Angular 2 and is loaded before
```
Could we replace many of these with https://polyfill.io/

gurmukhp replied:
Good idea, updated to use this service.

mhevery commented on src/styles.css:
```diff
@@ -0,0 +1,2 @@
+/* You can add global styles to this file, and also import other style files */
+@import '~@angular/material/core/theming/prebuilt/deeppurple-amber.css';
```
why `~`? That does not seem right.

gurmukhp replied:
I was following the [Theming guide](https://github.com/angular/material2/blob/master/docs/theming.md) in Angular Material 2:

"If you're using Angular CLI, this is as simple as including one line in your style.css file:
@import '~@angular/material/core/theming/prebuilt/deeppurple-amber.css';"

mhevery replied:
Ok, I will check with @jelbourn

jelbourn replied:
That's from the angular-cli (and webpack underneath that). See https://github.com/webpack/css-loader/

jelbourn replied:
If you're using the angular-cli, yes.

mhevery commented:
@vicb I am good on margining this. Can you look over before we do that. Also can you comment on the status of the https://github.com/angular/periscope/tree/v2 and how this relates to v2.

@gurmukhp sorry I just found out that we had v2, which means you probably duplicated some of this work. Sorry.

gurmukhp commented:
@mhevery  Ah ok, it was a good learning experience nevertheless :)

vicb commented on README.md:
```diff
@@ -0,0 +1,31 @@
+
+This project was generated with [angular-cli](https://github.com/angular/angular-cli) version 1.0.0-beta.17.
+
+## Development server
```
I would prefer not to repeat angular-cli docs here. 
A link should be enough.

vicb commented:
We worked on v2 during the hackathon.

The goal of the v2 branch is to work on the most valuable features that will help you improve the process ASAP.

We thought that upgrading would be a mess and starting over from a blank project created with the CLI. The reasoning is that it would probably be easier to port the current few components to the v2 branch that trying to update the master branch.

Also @IgorMinar has been working on a new GH-Firebase sync - not sure if it's in the v2 branch yet or a separate project. @IgorMinar can you chime in ?

Happy to discuss what we've worked on in the v2 branch and what seems most important for the team to work on first next week.

Thanks,
Vic

mhevery commented:
@vicb given that @gurmukhp did all the work upgrading it, What do you think we should do? Should we merge his version in, or should we continue on `v2` branch? I am voting on merging his version, but I don't feel strongly either way.

vicb commented:
I don't know what's best:
- what is the diff of the two versions ?
- if there is much diffs, we should ask the CLI team.

But either of the two is probably good, it's easy enough to extract what is independent of the CLI to move it to the other branch.

vicb commented:
We should not block on this - if the PR is ready otherwise, it should be merged.

mhevery commented:
+1 for merging.

vicb commented:
@gurmukhp please update the "." commit comment before merging. Thanks.

For future commits, it would also be nice to follow angular comment format, see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines

gurmukhp commented:
@vicb I've squashed all commits into one and updated as per the Angular comment format.

I'm unable to merge (don't think I have the right access).

Thanks

vicb commented:
Thanks, just merge, I have change the commit message to remove "(global)"
